pss/pss_test.go

```diff
@@ -507,116 +510,128 @@ func TestMessageProcessing(t *testing.T) {
 
 	addr := make([]byte, 32)
 	addr[0] = 0x01
-	ps := newTestPss(privkey, network.NewKademlia(addr, network.NewKadParams()), NewParams())
+	ps := newTestPssStart(privkey, network.NewKademlia(addr, network.NewKadParams()), NewParams(), false)
 	defer ps.Stop()
-
-	// message should pass
-	msg := newPssMsg(&msgParams{})
-	msg.To = addr
-	msg.Expire = uint32(time.Now().Add(time.Second * 60).Unix())
-	msg.Payload = &whisper.Envelope{
-		Topic: [4]byte{},
-		Data:  []byte{0x66, 0x6f, 0x6f},
+	outboxCapacity := 2
+
+	processed := make([]*PssMsg, 0)
+	successC := make(chan struct{})
+	ps.outbox = newOutbox(outboxCapacity)
+	forward := func(msg *PssMsg) bool {
+		processed = append(processed, msg)
+		successC <- struct{}{}
+		return true
 	}
-	if err := ps.handle(context.TODO(), msg); err != nil {
-		t.Fatal(err.Error())
+	ps.forwardFunc = forward
+
+	ps.Start(nil)
+	err = ps.enqueue(testRandomMessage())
+	if err != nil {
+		t.Fatal("Message should have space for enqueuing")
 	}
-	tmr := time.NewTimer(time.Millisecond * 100)
-	var outmsg *outboxMsg
-	select {
-	case outmsg = <-ps.outbox:
-	case <-tmr.C:
-	default:
+	usedSlots := ps.outbox.len()
+	if usedSlots != 1 {
+		t.Fatal("Outbox len should be 1 before finish processing")
 	}
-	if outmsg != nil {
-		t.Fatalf("expected outbox empty after full address on msg, but had message %s", msg)
+	t.Log("Message enqueued", "Outbox len", ps.outbox.len())
+
+	<-successC
+	if len(processed) != 1 {
+		t.Fatal("There should be one message processed")
```

Comment:
"incorrect number of processed messages; expected 1, got %"

Reply:
FIxed in 1b91b66